### PR TITLE
Feature/morsesmalecomplex

### PR DESCRIPTION
**Pull request by martinfalk**
* Morse-Smale complex computation with TTK
* topological simplification now in background thread
* various minor fixes

### CODE REVIEW

**Comment by martinfalk** [comment]
Investigating why ttk produces different results in regression test.

File: topovis/topologytoolkit/src/processors/morsesmalecomplex.cpp [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+                ->dispatch<std::shared_ptr<topology::MorseSmaleComplexData>,
+                           dispatching::filter::Scalars>([inportData](const auto buffer) {
+                    using ValueType = util::PrecisionValueType<decltype(buffer)>;
+                    using PrimitiveType = typename DataFormat<ValueType>::primitive;
+
+                    std::vector<int> offsets(inportData->getOffsets());
+
+                    ttk::MorseSmaleComplex morseSmaleComplex;
+                    morseSmaleComplex.setupTriangulation(
+                        const_cast<ttk::Triangulation*>(&inportData->getTriangulation()));
```
Comment (petersteneteg) [inline topovis/topologytoolkit/src/processors/morsesmalecomplex.cpp]: Was this fixed?

File: topovis/topologytoolkit/src/processors/morsesmalecomplextomesh.cpp [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    // Add critical points with their color
+    for (ttk::SimplexId i = 0; i < numcp; i++) {
+        positions[i].x = pMSCData->criticalPoints.points[3 * i];
+        positions[i].y = pMSCData->criticalPoints.points[3 * i + 1];
+        positions[i].z = pMSCData->criticalPoints.points[3 * i + 2];
+
+        switch (pMSCData->criticalPoints.cellDimensions[i]) {
+            case 0: {
+                colors[i] = propColors_.localMinima_.get();
+                // LogInfo("Min");
```
Comment (petersteneteg) [inline topovis/topologytoolkit/src/processors/morsesmalecomplextomesh.cpp]: Remove

File: topovis/topologytoolkit/src/processors/morsesmalecomplextomesh.cpp [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    std::vector<vec4> colors(numcp, {1.0f, 1.0f, 1.0f, 1.0f});
+    std::vector<float> radius(numcp, sphereRadius_.get());
+
+    // Add critical points with their color
+    for (ttk::SimplexId i = 0; i < numcp; i++) {
+        positions[i].x = pMSCData->criticalPoints.points[3 * i];
+        positions[i].y = pMSCData->criticalPoints.points[3 * i + 1];
+        positions[i].z = pMSCData->criticalPoints.points[3 * i + 2];
+
+        switch (pMSCData->criticalPoints.cellDimensions[i]) {
```
Comment (petersteneteg) [inline topovis/topologytoolkit/src/processors/morsesmalecomplextomesh.cpp]: This switch could be a function in propcolor

File: topovis/topologytoolkit/src/processors/morsesmalecomplextomesh.cpp [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    // - separatrixCells
+    mesh->addIndicies(Mesh::MeshInfo(DrawType::Lines, ConnectivityType::Strip),
+                      util::makeIndexBuffer(std::move(sepIndices)));
+    // vertex positions are already transformed
+    mesh->setModelMatrix(mat4(1.0f));
+    mesh->setWorldMatrix(pMSCData->triangulation->getWorldMatrix());
+    mesh->copyMetaDataFrom(*pMSCData->triangulation);
+
+    // enable primitive restart so we need only a single index buffer for multiple lines
+    glPrimitiveRestartIndex(0xffffffff);
```
Comment (petersteneteg) [inline topovis/topologytoolkit/src/processors/morsesmalecomplextomesh.cpp]: This is quite a hack might explain that... does this module depend on gl?

File: topovis/topologytoolkit/src/processors/morsesmalecomplex.cpp (hunk 1 above)
Comment (martinfalk) [reply topovis/topologytoolkit/src/processors/morsesmalecomplex.cpp]: constness was only partially fixed. See e.g.
`int ttk::MorseSmaleComplex::setInputScalarField(void *const data);`

**Comment by martinfalk** [comment]
rebased into #38

**Pull request closed without merging**